cluster/v1beta1/helpers.go

```diff
@@ -0,0 +1,119 @@
+package v1beta1
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	v1 "open-cluster-management.io/api/cluster/v1"
+)
+
+const clusterSetLabel = "cluster.open-cluster-management.io/clusterset"
+
+type ManagedClusterLister interface {
+	List(selector labels.Selector) (ret []*v1.ManagedCluster, err error)
+}
+
+type ManagedClusterSetLister interface {
+	List(selector labels.Selector) (ret []*ManagedClusterSet, err error)
+}
+
+// GetClustersFromClusterSet return the ManagedClusterSet's managedClusters
+func GetClustersFromClusterSet(clusterSet *ManagedClusterSet, clusterLister ManagedClusterLister) ([]*v1.ManagedCluster, error) {
+	var clusterSelector labels.Selector
+	var err error
+	var clusters []*v1.ManagedCluster
+
+	if clusterSet == nil {
+		return clusters, nil
+	}
+
+	clusterSelector, err = BuildClusterSelector(clusterSet)
+	if err != nil {
+		return clusters, err
+	}
+	if clusterSelector == nil {
+		return clusters, fmt.Errorf("failed to build ClusterSelector with clusterSet: %v", clusterSet)
+	}
+	clusters, err = clusterLister.List(clusterSelector)
+	if err != nil {
+		return clusters, fmt.Errorf("failed to list ManagedClusters: %w", err)
+	}
+	return clusters, nil
+}
+
+// GetClusterSetsOfClusterByClusterName return the managedClusterSets of a managedCluster using clusterName
+func GetClusterSetsOfClusterByClusterName(clusterName string, clusterSetLister ManagedClusterSetLister, clusterLister ManagedClusterLister) ([]*ManagedClusterSet, error) {
+	var err error
+	var returnClusterSets []*ManagedClusterSet
+
+	if len(clusterName) == 0 {
+		return returnClusterSets, nil
+	}
+	allClusterSets, err := clusterSetLister.List(labels.Everything())
+	if err != nil {
+		return returnClusterSets, err
+	}
+
+	for _, set := range allClusterSets {
+		setclusters, err := GetClustersFromClusterSet(set, clusterLister)
+		if err != nil {
+			return returnClusterSets, err
+		}
+		for _, sc := range setclusters {
+			if sc.Name == clusterName {
+				returnClusterSets = append(returnClusterSets, set)
+				break
+			}
+		}
+	}
+	return returnClusterSets, err
+}
+
+// GetClusterSetsOfClusterByClusterLabels return the managedClusterSets of a managedCluster using cluster labels
+func GetClusterSetsOfClusterByClusterLabels(cluster *v1.ManagedCluster, clusterSetLister ManagedClusterSetLister) ([]*ManagedClusterSet, error) {
```

Comment:
I think we need GetClusterSetsOfClusterByCluster which understand clusterselector type. GetClusterSetsOfClusterByClusterName seems not necessary.

Reply:
done